src/lib/list/selection-list.ts

```diff
@@ -394,6 +394,12 @@ export class MatSelectionList extends _MatSelectionListMixinBase implements Focu
         keyCode === HOME ? manager.setFirstItemActive() : manager.setLastItemActive();
         event.preventDefault();
         break;
+      case A:
+        if (event.ctrlKey) {
+          this.options.find(option => !option.selected) ? this.selectAll() : this.deselectAll();
```

Comment:
Is the extra check necessary? Looks like `_setAllOptionsSelected` already checks the existing state before considering the item "changed"

Reply:
We need to know whether there's at least one unselected option. The intended functionality is "deselect all if all are selected, otherwise select all".